VoodooInput/VoodooInputSimulator/VoodooInputSimulatorDevice.cpp

```diff
@@ -218,65 +218,53 @@ void VoodooInputSimulatorDevice::constructReportGated(const VoodooInputEvent& mu
     else
         input_report.TouchActive = 0x2;
     
-    int total_report_len = (9 * multitouch_event.contact_count) + 12;
-    IOBufferMemoryDescriptor* buffer_report = IOBufferMemoryDescriptor::inTaskWithOptions(kernel_task, 0, total_report_len);
-
     if (!is_error_input_active) {
-      buffer_report->writeBytes(0, &input_report, total_report_len);
-      handleReport(buffer_report, kIOHIDReportTypeInput);
+        handleReport(memDesc, kIOHIDReportTypeInput);
     }
-
-    buffer_report->release();
-    buffer_report = NULL;
-
-    if (!input_active) {
-        for (int i = 0; i < 15; i++) {
-            touch_state[i] = 2;
-        }
-
-        input_report.FINGERS[0].Size = 0x0;
-        input_report.FINGERS[0].Pressure = 0x0;
-        input_report.FINGERS[0].Touch_Major = 0x0;
-        input_report.FINGERS[0].Touch_Minor = 0x0;
-
-        milli_timestamp += 10;
-        
-        input_report.timestamp_buffer[0] = (milli_timestamp << 0x3) | 0x4;
-        input_report.timestamp_buffer[1] = (milli_timestamp >> 0x5) & 0xFF;
-        input_report.timestamp_buffer[2] = (milli_timestamp >> 0xd) & 0xFF;
-        
-        buffer_report = IOBufferMemoryDescriptor::inTaskWithOptions(kernel_task, 0, total_report_len);
-        buffer_report->writeBytes(0, &input_report, total_report_len);
-        handleReport(buffer_report, kIOHIDReportTypeInput);
-        buffer_report->release();
-        
-        input_report.FINGERS[0].Priority = 0x5;
-        input_report.FINGERS[0].State = 0x0;
-        
-        buffer_report = IOBufferMemoryDescriptor::inTaskWithOptions(kernel_task, 0, total_report_len);
-        buffer_report->writeBytes(0, &input_report, total_report_len);
-        handleReport(buffer_report, kIOHIDReportTypeInput);
-        buffer_report->release();
-        
-        milli_timestamp += 10;
-        
-        input_report.timestamp_buffer[0] = (milli_timestamp << 0x3) | 0x4;
-        input_report.timestamp_buffer[1] = (milli_timestamp >> 0x5) & 0xFF;
-        input_report.timestamp_buffer[2] = (milli_timestamp >> 0xd) & 0xFF;
-
-        buffer_report = IOBufferMemoryDescriptor::inTaskWithOptions(kernel_task, 0, 12);
-        buffer_report->writeBytes(0, &input_report, 12);
-        handleReport(buffer_report, kIOHIDReportTypeInput);
-        buffer_report->release();
-    }
-    
     input_report = {};
+    return;
+// The following code didn't have any impact in my unit.I don't think they are necessary.further testing needed.So,i'm just commenting them out.
```

Comment:
Please revert this as it may not effect your device, but IT WILL most certainly effect other devices.

Reply:
> Please revert this as it may not effect your device, but IT WILL most certainly effect other devices.

These additional reports sent to "hidd" when lifting finger off  the trackpad is ignored by the OS.This is more of a OS specific than a device specific.That being said,it needs further testing if we really need those aditional 3 Reports in the end.I'll revert this anyways just in case.

Reply:
I can assure you that the MT stack does need the extra input. When we (alexandred, coolstar and myself) were creating the initial implementation this was required. Otherwise, MT stack will not properly track gestures due.

The real MT2 device does this and best to leave as it is.